File: 숫자문자열과영단어/20221028/javascript/src/solution.js
const map = {
  zero: '0',
  one: '1',
  two: '2',
  three: '3',
  four: '4',
  five: '5',
  six: '6',
  seven: '7',
  eight: '8',
  nine: '9',
};

function test() {
  console.log(`replaceCharacter('one4'): 14 = ${replaceCharacter('one4')}`);
  console.log(`replaceCharacter('oneoneoneone'): 1111 = ${replaceCharacter('oneoneoneone')}`);
}

function solution(s) {
  // test();
  const answer = replaceCharacter(s);
  return answer;
}

function replaceCharacter(string) {
  return parseInt(Object.keys(map)
    .reduce((acc, key) => acc.replaceAll(key, map[key]), string), 10);
}

File: 숫자문자열과영단어/20221028/java/src/Solution.java
import java.util.*;

class Solution {
    public void test() {
        System.out.println("replaceCharacter(\"one4\"): 14 = " +
                replaceCharacter("one4"));
        System.out.println("replaceCharacter(\"one4seveneight\"): 1478 = " +
                replaceCharacter("one4seveneight"));
    }

    public int solution(String s) {
        // test();
        int answer = replaceCharacter(s);
        return answer;
    }

    public int replaceCharacter(String string) {
        Map<String, String> map = new HashMap<>();
        map.put("zero", "0");
        map.put("one", "1");
        map.put("two", "2");
        map.put("three", "3");
        map.put("four", "4");
        map.put("five", "5");
        map.put("six", "6");
        map.put("seven", "7");
        map.put("eight", "8");
        map.put("nine", "9");

        String newString = map.keySet().stream()
                .reduce(string, (acc, key) -> acc.replace(key, map.get(key)));

        return Integer.valueOf(newString);
    }
}
